live/root/usr/lib/dracut/modules.d/99agama-cmdline/agama-network-compat.sh

```diff
@@ -0,0 +1,128 @@
+#!/bin/bash
+
+[ -e /dracut-state.sh ] && . /dracut-state.sh
+
+. /lib/dracut-lib.sh
+. /lib/net-lib.sh
+
+ifcfg_to_ip() {
+  local ip
+  local v="${2}",
+  local interface="$1"
+  local conf_path="/etc/cmdline.d/40-agama-network.conf"
+  set --
+  while [ -n "$v" ]; do
+    set -- "$@" "${v%%,*}"
+    v=${v#*,}
+  done
+
+  if [[ $# -eq 0 ]]; then
+    echo "IFCFG 0 options given, must be wrong"
+    return 1
+  fi
+
+  ### See https://en.opensuse.org/SDB:Linuxrc#Network_Config
+  # ifcfg=<interface_spec>=[try,]dhcp*,[rfc2132,]OPTION1=value1,OPTION2=value2...
+  if str_starts "$1" "dhcp"; then
+    autoconf="$1"
+    if [ "$autoconf" = "dhcp4" ]; then
+      echo "AUTOCONF"
+      autoconf="dhcp"
+    fi
+    case $autoconf in
+    "dhcp" | "dhcp6")
+      if [ "$interface" = "*" ]; then
+        echo "ip=${1}" >>"/etc/cmdline.d/40-agama-network.conf"
+      else
+        echo "ip=${interface}:${1}" >>$conf_path
+      fi
+      ;;
+    *)
+      echo "No supported option ${1}"
+      ;;
+    esac
+
+    return 0
+  fi
+
+  # ifcifg=<interface_spec>=ip,gateway,nameserver,domain
+  if strglob "$1" "*.*.*.*/*"; then
+    [[ -n "$2" ]] && gateway=$2
+    [[ -n "$3" ]] && nameserver=$3
+
+    ip="$1 "
+    set --
+    while [ -n "$ip" ]; do
+      set -- "$@" "${ip%% *}"
+      ip="${ip#* }"
+    done
+
+    ## TODO: IP is a LIST_IP
+    ip="$1"
+    mask=${ip##*/}
+    ip=${ip%%/*}
+    shift
+
+    ## Configure the first interface, the gateway must belong to the same network
+    echo "ip=${ip}::${gateway}:$mask::${interface}" >>$conf_path
+
+    ## Configure multiple addresses for the same interface
+    while [[ $# -gt 0 ]]; do
+      ip="$1"
+      mask=${ip##*/}
+      ip=${ip%%/*}
+      echo "ip=${ip}:::$mask::${interface}" >>$conf_path
+      shift
+    done
+
+    ## Configure nameservers
+    if [[ -n $nameserver ]]; then
+      nameserver="$nameserver "
+      while [ -n "$nameserver" ]; do
+        echo "nameserver=${nameserver%% *}" >>$conf_path
+        nameserver="${nameserver#* }"
+      done
+    fi
```

Comment:
I have to say that without automatic testing I would be very scared to touch this file to avoid breaking anything.